integration-tests/actions/vrf/vrfv2/contract_steps.go

```diff
@@ -778,28 +468,130 @@ func RequestRandomnessAndWaitForFulfillment(
 	randomnessRequestCountPerRequestDeviation uint16,
 	randomWordsFulfilledEventTimeout time.Duration,
 ) (*vrf_coordinator_v2.VRFCoordinatorV2RandomWordsFulfilled, error) {
-	logRandRequest(l, consumer.Address(), coordinator.Address(), subID, minimumConfirmations, callbackGasLimit, numberOfWords, randomnessRequestCountPerRequest, randomnessRequestCountPerRequestDeviation)
-	_, err := consumer.RequestRandomness(
-		vrfKeyData.KeyHash,
+	logRandRequest(
+		l,
+		consumer.Address(),
+		coordinator.Address(),
 		subID,
 		minimumConfirmations,
 		callbackGasLimit,
 		numberOfWords,
 		randomnessRequestCountPerRequest,
+		randomnessRequestCountPerRequestDeviation,
+		vrfKeyData.KeyHash,
 	)
-	if err != nil {
-		return nil, fmt.Errorf("%s, err %w", vrfcommon.ErrRequestRandomness, err)
+	ch := make(chan *vrf_coordinator_v2.VRFCoordinatorV2RandomWordsFulfilled)
+	errorChannel := make(chan error)
+	go func() {
+		_, err := consumer.RequestRandomness(
+			vrfKeyData.KeyHash,
+			subID,
+			minimumConfirmations,
+			callbackGasLimit,
+			numberOfWords,
+			randomnessRequestCountPerRequest,
+		)
+		if err != nil {
+			l.Error().Err(err).Msg(err.Error())
+			errorChannel <- err
+		}
+	}()
+	go func() {
+		fulfillmentEvents, err := WaitForRequestAndFulfillmentEvents(
+			consumer.Address(),
+			coordinator,
+			vrfKeyData,
+			subID,
+			randomWordsFulfilledEventTimeout,
+			l,
+		)
+		if err != nil {
+			l.Error().Err(err).Msg("error waiting for RandomnessRequested and RandomWordsFulfilled events")
+			errorChannel <- err
+		}
+		ch <- fulfillmentEvents
+	}()
+	for {
+		select {
+		case err := <-errorChannel:
+			return nil, err
+		case fulfillmentEvent := <-ch:
+			return fulfillmentEvent, nil
+		case <-time.After(randomWordsFulfilledEventTimeout):
```

Comment:
Maybe try adding a small "delta" in timeout here since you have to account for time wasted for the `WaitForRequestAndFulfillmentEvents` bootstrapping before it starts executing. Maybe a second or two, if the timeout is something like 60s.

Reply:
not sure if I completely understand your point here - so randomWordsFulfilledEventTimeout is accounting for getting the final fulfilment event and it is provided from toml config and it should be big enough. Just dont think that a delta is needed here. Also, each chain would need to have this timeout customized.

Reply:
I see, so this is already taken into account by providing a sufficient value in the TOML config. All good then.